api/v1/listing.py

```diff
@@ -137,3 +140,67 @@ def delete(self):
             return {'message': response }, 500
         else:
             return {'message': response}, 200
+
+    def elapsed_time(self, start, end):
+        """
+        Return a reasonable number for elapsed time
+        :param start: datetime
+        :param end: datetime
+        :return: elapsed: string
+        """
+        duration = end - start
+        if duration.days == 0:
+            if duration.seconds > 3600:
+                return f'~{duration.seconds // 3600} hours'
+            elif duration.seconds > 60:
+                return f'~{duration.seconds // 60} minutes'
+            else:
+                return f'~{duration.seconds} seconds'
+        else:
+            return f'{duration.days} days'
+
+    def obfuscate_listing(self, listing):
+        """
+        Obfuscate the data returned from the db before presenting to the client
+        Fields returned:
+        listing: string: hash of the listing (a.k.a. id)
+        num_data_points: int: number of data points in the listing
+        avg_velocity: float: average velocity across all data points
+        duration: string: elapsed time between first and last data point
+        start_location: string: City, State location of the first data point
+        end_location: string: City, State location of the last data point
+        :param listing: dict: a Drive listing
+        :return: obfuscated: dict: an obfuscated listing
+        """
+        obfuscated = {}
+        obfuscated['listing'] = listing['listing']
+        obfuscated['num_data_points'] = len(listing['data'])
+        sum_velocity = 0
+        velocity_points = 0
+        start_timestamp = datetime.now()
+        end_timestamp = datetime(1970, 1, 1, 12, 0, 0)
+        start_location = None
+        end_location = None
+        for location in listing['data']:
+            if 'velocity' in location:
+                sum_velocity = sum_velocity + float(location['velocity'])
```

Comment:
can we not use `+=` on these 2 vars? guess it doesn't really matter...

Reply:
Yeah. Updated. `+=` is cleaner.